packages/discovery-provider/src/tasks/entity_manager/entities/playlist.py

```diff
@@ -267,6 +270,73 @@ def validate_playlist_tx(params: ManageEntityParameters):
             and params.metadata.get("is_private")
         ):
             raise IndexingValidationError(f"Cannot unlist playlist {playlist_id}")
+        validate_access_conditions(params)
+
+
+def validate_access_conditions(params: ManageEntityParameters):
+    playlist_metadata = params.metadata
+
+    is_stream_gated = playlist_metadata.get("is_stream_gated")
+    stream_conditions = playlist_metadata.get("stream_conditions", {}) or {}
+
+    if is_stream_gated:
+        # if stream gated, must have stream conditions
+        if not stream_conditions:
+            raise IndexingValidationError(
+                f"Playlist {params.entity_id} is stream gated but has no stream conditions"
+            )
+        # must be gated on a single condition
+        if len(stream_conditions) != 1:
+            raise IndexingValidationError(
+                f"Playlist {params.entity_id} has an invalid number of stream conditions"
+            )
+
+
+# Make sure that access conditions do not incorrectly change during track update.
+# Rule of thumb is that access can only be modified to decrease strictness.
+def validate_update_access_conditions(params: ManageEntityParameters):
+    playlist_id = params.entity_id
+    if playlist_id not in params.existing_records["Playlist"]:
+        raise IndexingValidationError(
+            f"Playlist {playlist_id} is not in existing records"
+        )
+
+    existing_playlist = helpers.model_to_dictionary(
+        params.existing_records["Playlist"][playlist_id]
+    )
+    # TODO: scheduled release playlists are not restricted on how they can be updated
+    # if existing_playlist.get("is_scheduled_release"):
```

Comment:
should these comments not be removed if the the TODO comment right above is true?